### PR TITLE
Add transaction history components

### PR DESCRIPTION
**Pull request by desfero**
**Testing notes**
This pr adds the UI for transaction history in wallet #4248 
IOS: https://i.diawi.com/e2EzS3
Android: https://i.diawi.com/R23kpf

### CODE REVIEW

File: packages/wallet/app/config/externalRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { withParams } from "@neufund/shared-utils";
+
+export const externalRoutes = {
+  platformWalletConnect: "platform.neufund.org/wc",
```
Comment (arjunu) [inline packages/wallet/app/config/externalRoutes.ts:4]: `https://` ?

File: packages/wallet/app/assets/pending.svg [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none" viewBox="0 0 24 24">
+    <g stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" clip-path="url(#prefix__clip0)">
+        <path d="M12 .747c6.213 0 11.25 5.037 11.25 11.25S18.213 23.247 12 23.247.75 18.21.75 11.997 5.787.747 12 .747z"/>
+        <path d="M8.252 17.612c-3.101-2.07-3.938-6.261-1.868-9.363.96-1.44 2.437-2.455 4.125-2.837M18.3 14.394c-.34.903-.872 1.72-1.558 2.4M14.058 18.425c-.919.295-1.891.388-2.849.272"/>
+        <path d="M12 3.747c.828 0 1.5.672 1.5 1.5s-.672 1.5-1.5 1.5-1.5-.672-1.5-1.5.672-1.5 1.5-1.5z"/>
+    </g>
+    <defs>
+        <clipPath id="prefix__clip0">
```
Comment (arjunu) [inline packages/wallet/app/assets/pending.svg:8]: I've faced problems with two inline SVGs having same IDs. So this prefix should be made unique. Currently it's used in wallet.svg & home.svg

File: packages/wallet/app/components/shared/Icon.tsx [hunk 3]
```diff
@@ -47,18 +48,10 @@ const icons = {
   [EIconType.SHARE]: Share,
   [EIconType.WALLET]: Wallet,
   [EIconType.YES]: Yes,
+  [EIconType.PENDING]: Pending,
```
Comment (arjunu) [inline packages/wallet/app/components/shared/Icon.tsx:51]: this is alphabetically sorted

File: packages/wallet/app/config/externalRoutes.ts (hunk 1 above)
Comment (arjunu) [reply packages/wallet/app/config/externalRoutes.ts:4]: I see this isn't used anywhere

**Comment by arjunu** [comment]
Doesn't work on Android:
image

File: packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+              )}
+              numberOfLines={1}
+            >
+              {direction === ETransactionDirection.OUT && <>&minus;</>}
+              <Money currency={valueToken} value={value} decimalPlaces={valueDecimals} />
+            </TextBold>
+
+            {valueEquivalent && valueEquivalentToken && valueEquivalentDecimals && (
+              <HelperText style={styles.valueEquivalent} numberOfLines={1}>
+                &asymp;{" "}
```
Comment (arjunu) [inline packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx:102]:
```suggestion
                &asymp;&nbsp;
```

File: packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+            <View style={styles.dateAndBadgeWrapper}>
+              <HelperText style={styles.date} numberOfLines={1}>
+                <FormattedDate value={timestamp} year="numeric" month="short" day="numeric" />
+              </HelperText>
+              <Badge type={EBadgeType.PENDING}>Pending</Badge>
+            </View>
+          </View>
+
+          <View>
```
Comment (arjunu) [inline packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx:88]: this section looks same as the one in SuccessfulTransation, perhaps can be shared?

File: packages/wallet/app/config/externalRoutes.ts (hunk 1 above)
Comment (desfero) [reply packages/wallet/app/config/externalRoutes.ts:4]: done

File: packages/wallet/app/assets/pending.svg (hunk 2 above)
Comment (desfero) [reply packages/wallet/app/assets/pending.svg:8]: interesting was the id clash happening in react-native or web? (also fixed)

File: packages/wallet/app/components/shared/Icon.tsx (hunk 3 above)
Comment (desfero) [reply packages/wallet/app/components/shared/Icon.tsx:51]: done

File: packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx (hunk 4 above)
Comment (desfero) [reply packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx:102]: done

Comment (desfero) [reply packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx:102]: btw, if you are curious `&asymp;` and `&minus;` are used here for better accessibility over (~ and -)

File: packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx (hunk 5 above)
Comment (desfero) [reply packages/wallet/app/components/wallet/transactions/PendingTransaction.tsx:88]: I would say let's keep it not shared as it looks like a hasty abstraction in a long term

File: packages/wallet/app/assets/pending.svg (hunk 2 above)
Comment (arjunu) [reply packages/wallet/app/assets/pending.svg:8]: On web, so not sure about react native

**Comment by glupiawymowka** [comment]
- [ ] it should always be nEUR, not nEur
image
image

**Pull request merged**